src/components/screentock.js

```diff
@@ -14,44 +14,116 @@
+      const ctxScreenshot = screenshotCanvas.getContext("2d");
+
+      // draw image from Aframe canvas to screenshot canvas
+      ctxScreenshot.drawImage(aframeCanvas, 0, 0);
+      // add scene title to screenshot
+      addTitleToCanvas(ctxScreenshot, screenshotCanvas.width, screenshotCanvas.height);
+      // add 3DStreet logo
+      await addLogoToCanvas(ctxScreenshot);
+      return screenshotCanvas;
+    }
+
+    function addTitleToCanvas (ctx, screenWidth, screenHeight) {
+      ctx.font = "25px Lato";
+      ctx.textAlign = 'center';
+      ctx.fillStyle = '#FFF';
+      ctx.fillText(STREET.utils.getCurrentSceneTitle(), 
+        screenWidth - screenWidth/2, 
+        screenHeight - 43);
+    }
+
+    const addLogoToCanvas = async (ctx) => {
+      
+      const logoImg = document.querySelector('img.viewer-logo-img') 
+      const logoSVG = document.querySelector('#aframeInspector #logoImg svg');
+
+      if (logoImg) {
+        ctx.drawImage(logoImg, 0, 0, 135, 43, 40, 30, 135, 43);
+      } else if (logoSVG) {
+        const image = new Image();
+        image.src = `data:image/svg+xml;base64,${window.btoa(logoSVG.outerHTML)}`;
+        await new Promise((resolve) => {
+          image.onload = resolve;
+        });
+
+        ctx.drawImage(image, 0, 0, 135, 23, 40, 40, 135, 23);
+      }      
+    }
+
+    function downloadImageDataURL (filename, dataURL, scnrenshotCanvas) {
+      const element = document.createElement('a');
+      const url = dataURL.replace(/^data:image\/[^;]/, 'data:application/octet-stream');
+      element.setAttribute('href', url);
+      element.setAttribute('download', filename);
+      element.style.display = 'none';
+      document.body.appendChild(element);
+      element.click();
+      document.body.removeChild(element);
+    }
+
+    const saveFilename = filename + '.' + type;
+    
+    // render one frame
+    renderer.render(AFRAME.scenes[0].object3D, AFRAME.scenes[0].camera);
+    const screenshotCanvas = await createCanvasWithScreenshot(renderer.domElement);
+
+    if (type == 'img') {
+      imgElement.src = renderer.domElement.toDataURL();
+      return;
+    }
+    if (type == 'png') {
+      downloadImageDataURL(saveFilename, screenshotCanvas.toDataURL('image/png'));
+    } else {
+      downloadImageDataURL(saveFilename, screenshotCanvas.toDataURL('image/jpeg', 0.95));
+    }
+    // show helpers
+    toggleHelpers(true);
+  },
+  update: function (oldData) {
+    // If `oldData` is empty, then this means we're in the initialization process.
+    // No need to update.
+    if (Object.keys(oldData).length === 0) { return; }
+    
     // this should be paused when not in use. could be throttled too
     if (this.data.takeScreenshot) {
-      this.el.setAttribute('screentock', 'takeScreenshot', false);
-      takeScreenshotNow(this.data.filename, this.data.type, this.data.imgElementSelector);
-    }
+      this.data.takeScreenshot = false;
```

Comment:
Hi @Algorush was there any specific reason to change this
`      this.el.setAttribute('screentock', 'takeScreenshot', false);`
into this
`       this.data.takeScreenshot = false;`

In general we're not supposed to do that with a-frame, always using `setAttribute` so that lifecycle methods / events are properly triggered, but maybe you needed to in this case for some reason?

Reply:
through setAttribute the update event is triggered once again. In this case it is not necessary, in other cases it is different. Here takeScreenshot is used only inside the component